src/txn/txn_ckpt.c

```diff
@@ -580,11 +582,30 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, const char *cfg[])
 	txn_global = &conn->txn_global;
 	txn_state = WT_SESSION_TXN_STATE(session);
 
+	/*
+	 * Someone giving us a specific timestamp overrides the general
+	 * use_timestamp.
+	 */
 	WT_RET(__wt_config_gets(session, cfg, "read_timestamp", &cval));
 	if (cval.len > 0) {
 		WT_RET(__wt_snprintf(timestamp_config, sizeof(timestamp_config),
 		    "read_timestamp=%.*s", (int)cval.len, cval.str));
 		txn_cfg[2] = timestamp_config;
+	} else if (txn_global->has_oldest_timestamp) {
+		WT_RET(__wt_config_gets(session, cfg, "use_timestamp", &cval));
```

Comment:
I'm concerned about the order of checking conditions in here. If an application wants to create a checkpoint at a timestamp, but there is no oldest_timestamp set (maybe it's creating a checkpoint shortly after startup?), then we'll create a checkpoint that includes all committed changes. Is that a non-issue because as soon as we make an update that has an associated timestamp, we are guaranteed that `has_oldest_timestamp` is true? I don't think that's true, since setting a commit timestamp doesn't force refreshing the global transaction state which is required to set the global timestamps.